File: 0207-course-schedule/0207-course-schedule.cpp
class Solution {
public:
    bool canFinish(int numCourses, vector<vector<int>>& prerequisites) {
        vector<int> inDegrees(numCourses, 0);
        vector<vector<int>> graph(numCourses);
        for(int i=0;i<prerequisites.size();i++) {
            graph[prerequisites[i][1]].push_back(prerequisites[i][0]);
            inDegrees[prerequisites[i][0]]++;
        }

        for(int i=0;i<numCourses;i++) {
            int j=0;
            while(j<numCourses) {
                if(inDegrees[j]==0) {
                    break;
                }
                j++;
            }
            if(j==numCourses) return false;
            inDegrees[j] = -1;
            for(int k=0;k<graph[j].size();k++) {
                inDegrees[graph[j][k]]--;
            }
        }
        return true;
    }
};

File: 0207-course-schedule/0207-course-schedule.js
/**
 * @param {number} numCourses
 * @param {number[][]} prerequisites
 * @return {boolean}
 */
var canFinish = function(numCourses, prerequisites) {
    let indegree = Array(numCourses).fill(0), graph = new Map();

    for(let i=0;i<numCourses;i++) {
        graph[i] = [];
    }    
    for(let i=0;i<prerequisites.length;i++) {
        graph[prerequisites[i][1]].push(prerequisites[i][0]);
        indegree[prerequisites[i][0]]++;
    }    
    for(let i=0;i<numCourses;i++) {
        let j = 0;
        while(j<numCourses) {
            if(indegree[j]==0) {
                break;
            }
            j++;
        }
        if(j==numCourses) return false;
        indegree[j]=-1;
        for(let k = 0;k<graph[j].length;k++) {
            indegree[graph[j][k]]--;
        }
    }
    return true;

};